ucla_migration.drush.inc

```diff
@@ -3830,18 +3843,32 @@ EOQ;
         // get mods datastream from object if exist or else create a new one add this logic
         watchdog('Ingest Success', 'In add teixml @file_name', array('@file_name' => $tei_path), WATCHDOG_NOTICE);
         $object->relationships->add(FEDORA_MODEL_URI, 'hasModel', 'islandora:manuscriptCModel');
+        $is_new_mods_ds = FALSE;
         if (isset($object['TEI'])) {
             $datastream = $object['TEI'];
         } else {
             $datastream = $object->constructDatastream('TEI', 'M');
             $datastream->label = "TEI Record";
             $datastream->mimetype = "text/xml";
+            $is_new_mods_ds = TRUE;
         }
         watchdog('Ingest Success', 'Found datastream @file_name', array('@file_name' => $tei_path), WATCHDOG_NOTICE);
         //$datastream->setContentFromFile($mods_path);
-        $tei_file = file_get_contents($tei_path);
-        $datastream->content = $tei_file;
-        $object->ingestDatastream($datastream);
+        try {
+            $tei_file = file_get_contents($tei_path);
+            $datastream->content = $tei_file;
+            if ($is_new_mods_ds) {
+                $object->ingestDatastream($datastream);
+            }
+        }catch (Exception $e) {
```

Comment:
a nit pick, but the formatting is a little off in some places

Reply:
Fixed formatting. for the catch block